lib/Target/ExportSMTLIB/ExportSMTLIB.cpp

```diff
@@ -343,7 +357,44 @@ struct ExpressionVisitor
       info.stream << ")";
 
     if (weight != 0)
-      info.stream << " :weight " << weight << ")";
+      info.stream << " :weight " << weight;
+    if(!patterns.empty()){
+      info.stream << "\n :pattern (";
+      for(auto &p: patterns){
+
+        for (auto [i, arg] : llvm::enumerate(p.getArguments())) {
+
+          // retrieve argument name from the body region
+          info.valueMap.insert(arg, argNames[i].str());
+
+        }
+        SmallVector<Value> worklist;
+
+        Value yieldedValue = p.front().getTerminator()->getOperand(0);
+
+        worklist.push_back(yieldedValue);
+
+        unsigned indentExt = operatorString.size() + 2;
+
+        VisitorInfo newInfo(info.stream, info.valueMap,
```

Comment:
is the previous VisitorInfo called newInfo still in scope here? If so renaming it might make this slightly easier to read